### PR TITLE
Enable SAN on libraries built in separate cmake files

### PR DESCRIPTION
**Pull request by achamayou**
While looking at #1701, I noticed that we do not use the sanitiser when building some external libraries, which means that we potentially ignore some issues.

It's unlikely that we can fix all of them, but we ought to be aware of them at least, and chase the ones that have a serious impact.

### CODE REVIEW

File: cmake/common.cmake [hunk 1]
```diff
@@ -73,6 +73,7 @@ find_package(MbedTLS REQUIRED)
 set(CLIENT_MBEDTLS_INCLUDE_DIR "${MBEDTLS_INCLUDE_DIRS}")
 set(CLIENT_MBEDTLS_LIBRARIES "${MBEDTLS_LIBRARIES}")
 
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/tools.cmake)
```
Comment (eddyashton) [inline cmake/common.cmake:76]: Looks like this file isn't included in the PR.

**Comment by achamayou** [comment]
I think that's all of them, the only one that looks suspicious is the zero size memcpy happening in the MerkleTree in the snapshot recovery test, which @jumaffre is taking a look at.

This is otherwise locking in the current situation to avoid further regressions.

Separately, I will try further steps to fix the LeakSanitiser break (#1701).

**Comment by jumaffre** [comment]
The MerkleTree issue has been identified and is now tracked here: https://github.com/project-everest/hacl-star/issues/376

**Comment by eddyashton** [comment]
Ignoring unrelated failure in the Daily CI, merging anyway.

**Pull request merged**